boards/xtensa/m5stack_core2/m5stack_core2.dts

```diff
@@ -0,0 +1,237 @@
+	status = "disabled";
+};
+
+&uart0 {
+	status = "okay";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_tx_gpio1 &uart0_rx_gpio3>;
+	pinctrl-names = "default";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 21 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 22 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c0_default>;
+	scl-timeout-us = <0>;
+	pinctrl-names = "default";
+
+	pfc8563_rtc: pfc8563@51 {
+		compatible = "nxp,pcf8523";
+		reg = <0x51>;
+		status = "okay";
+		battery-switch-over = "disabled";
+	};
+
+	axp192_pmic: axp192@34 {
+		compatible = "x-powers,axp192";
+		reg = <0x34>;
+		status = "okay";
+
+		axp192_regulator: axp192_regulator {
+			compatible = "x-powers,axp192-regulator";
+			status = "okay";
+
+			vdd_mcu: DCDC1 {
+				regulator-init-microvolt = <3350000>;
+				regulator-min-microvolt = <3200000>;
+				regulator-max-microvolt = <3400000>;
+				regulator-initial-mode = <AXP192_DCDC_MODE_AUTO>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			lcd_bg: DCDC3 {
+				regulator-init-microvolt = <2800000>;
+				regulator-min-microvolt = <0>;
+				regulator-max-microvolt = <2800000>;
+				regulator-boot-on;
+			};
+
+			v_peri: LDO2 {
+				regulator-init-microvolt = <3300000>;
+				regulator-min-microvolt = <3200000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+			};
+
+			vib_motor: LDO3 {
+				regulator-init-microvolt = <2800000>;
+			};
+		};
+
+		axp192_gpio: axp192_gpio {
+			compatible = "x-powers,axp192-gpio";
+			gpio-controller;
+			#gpio-cells = <2>;
+			ngpios = <5>;
+			status = "okay";
+
+			pwr_led: axp192_gpio1 {
+				gpio-hog;
+				gpios = <1 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
+				output-high;
+				line-name = "pwr-led";
+			};
+
+			bus_pwr_en: axp192_gpio0 {
+				gpio-hog;
+				gpios = <0 0>;
+				input;
+			};
+		};
+	};
+
+	ft5336@38 {
+		compatible = "focaltech,ft5336";
+		reg = <0x38>;
+		int-gpios = <&gpio1 7 0>;
+
+		kscan_input: kscan-input {
+			compatible = "zephyr,kscan-input";
+		};
+	};
+};
+
+&spi3 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	pinctrl-0 = <&spim3_default>;
+	pinctrl-names = "default";
+	dma-enabled;
+	clock-frequency = <20000000>;
+	cs-gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
+
+	ili9342c: ili9342c@0 {
+		compatible = "ilitek,ili9342c";
+		spi-max-frequency = <30000000>;
+		reg = <0>;
+		cmd-data-gpios = <&gpio0 15 GPIO_ACTIVE_LOW>;
+		vin-supply = <&lcd_bg>;
+		reset-gpios = <&axp192_gpio 4 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
+		pixel-format = <ILI9XXX_PIXEL_FORMAT_RGB565>;
+		display-inversion;
+		width = <320>;
+		height = <240>;
+		rotation = <0>;
+	};
+};
+
+
+&timer0 {
+	status = "okay";
+};
+
+&timer1 {
+	status = "okay";
+};
+
+&timer2 {
+	status = "okay";
+};
+
+&timer3 {
+	status = "okay";
+};
+
+&trng0 {
+	status = "okay";
+};
+
+&flash0 {
+	status = "okay";
+	reg = <0 0x400000>;
```

Comment:
Please remove the `reg` property as it has incorrect size. Correct size is defined in the `esp32_wrover_e_n16r8.dtsi`.

Reply:
reg definition has been fixed.

Reply:
@marekmatej Are you ok with using `esp32_d0wd_v3` as the base?